junit/kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java

```diff
@@ -188,4 +195,39 @@ public void reset() {
       ((Resetable) this.dispatcher).reset();
     }
   }
+
+  /**
+   * Ensure that the server will supply an {@link APIResourceList} containing an {@link APIResource}
+   * representing the {@link CustomResourceDefinitionContext} from the apis/group/version endpoint.
+   * <p>
+   * This is useful when testing calls through the {@link KubernetesClient#genericKubernetesResources(String, String)}
+   * entry point.
+   * <p>
+   * If this is a crud server, the custom resource will be added to the set of previously added resources
+   * and the resources inferred from custom resource definitions that have been added.
+   * <p>
+   * If this server is not crud, this call will add a single expectation for the given resource. Direct handling of
+   * multiple resources for a given api group/version has not yet been added.
+   *
+   * @param rdc the resource definition context
+   */
+  public void expectCustomResource(CustomResourceDefinitionContext rdc) {
```

Comment:
Quick question, why doesn't this class implement the new `CustomResourceAware` interface?

Reply:
No reason, I've added that now.